content/docs/intro/concepts/programming-model.md

```diff
@@ -9,4529 +9,104 @@ menu:
 aliases: ["/docs/reference/programming-model/"]
 ---
 
-This page describes the Pulumi programming model. These are the core concepts used when authoring an infrastructure as code program. Each program describes a set of desired infrastructure resources; the Pulumi CLI knows how to run, preview, and deploy those resources, both for the initial provisioning, as well as subsequent updating.
+## Concepts {#concepts}
 
-## Introduction {#introduction}
+In this section, we introduce the basic concepts required to start declaring your infrastructure with Pulumi. Throughout the remainder of this page, we'll examine the ins and outs of Pulumi's programming model concepts.
 
-In this section, we will see the most basic concepts required to start declaring your infrastructure with Pulumi. Throughout the remainder of this page, we'll examine the ins and outs of Pulumi's programming model concepts.
+## Pulumi Programming Model {#pulumi-programming-model}
```

Comment:
I suggest removing this subheading per my earlier suggestion above about keeping this scoped to the programming model concepts and not talking about the deployment environment here.